lib/generation/generators/value_objects/generation_configuration/pb_generation_configuration.dart

```diff
@@ -97,9 +97,12 @@ abstract class GenerationConfiguration {
   }
 
   Future<void> _generateNode(PBIntermediateNode node, String filename) async {
-    await _generationManager.fileStrategy.generatePage(
-        await _generationManager.generate(await applyMiddleware(node)),
-        filename,
-        args: node is InheritedScaffold ? 'SCREEN' : 'VIEW');
+    if (node?.auxiliaryData?.stateGraph?.states?.isNotEmpty ?? false) {
```

Comment:
A big assumption I made here is that `applyMiddleware` would be in charge of writing the files necessary for state management. Because of this, I am no longer calling the `generatePage` method, since for both `provider` and `stateful` managements, we make sure that the appropriate files are generated within `applyMiddleware`, rendering the `generatePage` as it was being used before useless. Feel free to discuss with me if you have any doubts about the assumptions I made.